spec/requests/api/v1/users/update_spec.rb

```diff
@@ -0,0 +1,58 @@
+require 'rails_helper'
+
+describe 'PUT api/v1/users/:id', type: :request do
+  let(:user)              { create(:user) }
+  let(:new_email)         { 'test2@test.com' }
+  let(:new_name)          { 'Test2' }
+  let(:new_gender)        { 'female' }
+
+  context 'with valid params' do
+    let(:params) do
+      {
+        user: { email: new_email, name: new_name, gender: new_gender }
+      }
+    end
+
+    it 'returns a succesful response' do
+      put api_v1_user_path(user), params: params, headers: auth_headers, as: :json
```

Comment:
extract `put api_v1_user_path(user), params: params, headers: auth_headers, as: :json` to a before block